src/developer/core/plugin-api.md

```diff
@@ -69,16 +72,120 @@ This is the workhorse.
+
+### Plugin callback methods
+
+Callback methods are triggered by the enclosing application. These can be divided into
+three categories: Startup/cleanup, processing, and UI interactions.
+
+| Startup/cleanup callbacks          | Description                                                                                                                                                                                                                                                                                                                         |
+|------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `__init__(self, g_pool, **kwargs)` | Called when a plugin instance is started. `g_pool` provides access to the application. Calling `super().__init__(g_pool)` is strongly recommended. `kwargs` can be used for user preferences. See example below.                                                                                                                    |
+| `init_ui(self)`                    | Called after `__init__` if the calling process provides a user interface. Allows the plugin to setup its settings menu, quick access buttons, etc.                                                                                                                                                                                  |
+| `cleanup(self)`                    | Called when `Plugin.alive` is set to `True`, i.e. on application shutdown or if the plugin is being disabled                                                                                                                                                                                                                        |
+| `deinit_ui(self)`                  | Called before `cleanup` and the calling process provides a user interface. The plugin is responsible for removing any UI elements added in `init_ui`.                                                                                                                                                                               |
+| `get_init_dict(self)`              | Called on each active plugin instance on application shutdown. Returns a dictionary which is stored in the application's persistent session settings. On the next application launch, all previously active plugins will be restored by calling `__init__` and passing the dictionary as the `kwargs` arguments. See example below. |
+
+Session settings example:
+```python
+from plugin import Plugin
+
+class MyCustomPlugin(Plugin):
+    def __init__(self, g_pool, my_custom_setting=5):
+        self._my_custom_setting = my_custom_setting
```

Comment:
I like these code snippets. Can we include some notes in the code that show an example of what a user might want to save? E.g. are they saving GUI elements (slider values etc.) or other things such as custom functions run on gaze datums, or both?